service/models.py

```diff
@@ -182,21 +183,20 @@ def deserialize(self, data: dict):
             data (dict): A dictionary containing the resource data
         """
         try:
-            self.name = data["name"]
-            self.description = data["description"]
-            self.category = data["category"]
-            self.check_price(data["price"])
-            self.check_available(data["available"])
+            if "name" in data:
```

Comment:
Shouldn't we maintain the parameters such as :
 self.name = data["name"]
 self.description = data["description"]
 self.category = data["category"]
 self.check_price(data["price"])
 self.check_available(data["available"])
 
 I think these fields are required. 
 Consequently, if any of these required fields are missing, errors can be thrown using try-catch block

Reply:
The thing is both `create` and `update` use this `deserialize` function, and `update` don't need all fields to be required. Thus I deleted the checking in  `deserialize`, and added an error handler in `model.py/create`.